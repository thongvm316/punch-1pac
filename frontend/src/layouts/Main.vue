<template>
  <div>
    <main>
      <div class="container grid-xl">
        <announcements/>
        <remind-punch-in/>
        <h1 class="title">{{ title }}</h1>
        <slot></slot>
      </div>
    </main>
  </div>
</template>

<script>
const Announcements = () => import('../components/Announcements')
const RemindPunchIn = () => import('../components/RemindPunchIn')

export default {
  name: 'main-layout',

  props: {
    title: String,
    isOpenAnnoucement: Boolean
  },

  components: {
    Announcements,
    RemindPunchIn
  }
}
</script>
