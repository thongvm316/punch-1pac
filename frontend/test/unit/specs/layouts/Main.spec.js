import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import Main from '@/layouts/Main'

const scopedSlots = {
  default: '<p class="default-scoped">Dashboard page</p>'
}

Object.assign(wrapperOps, {
  propsData: {
    title: 'Dashboard'
  },
  stubs: {
    Announcements: true,
    RemindPunchIn: true
  },
  scopedSlots
})

const wrapper = shallowMount(Main, wrapperOps)

describe('Main.vue', () => {
  it('should render Main layout correctly', () => {
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
