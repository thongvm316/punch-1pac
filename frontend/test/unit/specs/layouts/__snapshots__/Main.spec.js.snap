// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main.vue should render Main layout correctly 1`] = `
<main>
  <div class="container grid-xl">
    <announcements-stub></announcements-stub>
    <remindpunchin-stub></remindpunchin-stub>
    <h1 class="title">
      Dashboard
    </h1>
    <p class="default-scoped">Dashboard page</p>
  </div>
</main>
`;
